packages/lesswrong/server/postgresView.ts

```diff
@@ -29,11 +35,21 @@ class PostgresView {
       await queryWithLock(db, this.refreshQuery, this.queryTimeout);
     }
   }
+
+  registerCronJob(addCronJob: (options: CronJobSpec) => void) {
+    addCronJob({
+      name: `refreshPostgresView-${this.name}`,
+      interval: this.refreshInterval,
+      job: () => this.refresh(getSqlClientOrThrow()),
+    });
+  }
 }
 
 const postgresViews: PostgresView[] = [];
 
 export const createPostgresView = (
+  name: string,
+  refreshInterval: string,
```

Comment:
Why is this a required argument? If the refreshQuery is optional? And like, if some views aren't materialized?

Reply:
Just because this was only designed to handle materialized views - I've just tidied it up to make in the interface nicer.